platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheProblemReportingIntegrationTest.groovy

```diff
@@ -272,6 +273,58 @@ class ConfigurationCacheProblemReportingIntegrationTest extends AbstractConfigur
         }
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/13862")
+    def "state deserialization errors always halt the build and includes the cause"() {
+        given:
+        def configurationCache = newConfigurationCacheFixture()
+
+        buildFile << """
+            class BrokenSerializable implements java.io.Serializable {
+                private Object readResolve() {
+                    throw new RuntimeException("BOOM")
+                }
+            }
+
+            class BrokenTaskType extends DefaultTask {
+                final prop = new BrokenSerializable()
+            }
+
+            task broken(type: BrokenTaskType)
+        """
+
+        when:
+        configurationCacheFails 'broken'
+
+        then:
+        failure.assertTasksExecuted()
+
+        and:
+        configurationCache.assertStateStored()
+        failure.assertHasFailures(1)
+        failure.assertHasFileName("Build file '${buildFile.absolutePath}'")
+        failure.assertHasLineNumber(4)
+        failure.assertHasDescription("Could not load the value of field `prop` of task `:broken` of type `BrokenTaskType`.")
+        failure.assertHasCause("BOOM")
+        problems.assertResultHasProblems(failure) {
```

Comment:
💅 We either should add a predicate here or drop the empty lambda

Reply:
Ah, good catch. I thought the lambda was required (as there is no default value for the parameter and other existing code provided a no-op closure, but only now I noticed a missing/null argument will be replaced by a no-op closure here: https://github.com/gradle/gradle/blob/9239189ef8d307b43b765f5b9637f509d8f5a436/platforms/core-configuration/model-core/src/main/java/org/gradle/util/internal/ConfigureUtil.java#L116